Library/Homebrew/cmd/list.sh

```diff
@@ -0,0 +1,92 @@
+#:  * `list`
+#:
+#: List all installed formulae and casks.
+#: If <formula> is provided, summarise the paths within its current keg.
+#: If <cask> is provided, list its artifacts.
+
+# HOMEBREW_CELLAR, HOMEBREW_PREFIX are set by brew.sh
+# shellcheck disable=SC2154
+
+homebrew-list() {
+  case "$1" in
+    # check we actually have list and not e.g. listsomething
+    list) ;;
+    list*) return 1 ;;
+    *) ;;
+  esac
+
+  local ls_args=()
+  local formula=""
+  local cask=""
+
+  while [[ "$#" -gt 0 ]]
+  do
+    case "$1" in
+      # check for flags passed to ls
+      -1 | -l | -r | -t)
+        ls_args+=("$1")
+        shift
+        ;;
+      --formula | --formulae)
+        formula=1
+        shift
+        ;;
+      --cask | --casks)
+        cask=1
+        shift
+        ;;
+      # reject all other flags
+      *) return 1 ;;
+    esac
+  done
+
+  local tty
+  if [[ -t 1 ]]
+  then
+    tty=1
+  fi
+
+  local error_string="LS_ERRORED"
+  if [[ -z "${cask}" && -d "${HOMEBREW_CELLAR}" ]]
+  then
+    if [[ -n "${tty}" && -z "${formula}" ]]
+    then
+      ohai "Formulae"
+    fi
+
+    local formula_output
+    formula_output="$(ls "${ls_args[@]}" "${HOMEBREW_CELLAR}" || echo "${error_string}")"
+    if [[ "${formula_output}" == "${error_string}" ]]
+    then
+      exit 1
+    elif [[ -n "${formula_output}" ]]
```

Comment:
```suggestion
    formula_output="$(ls "${ls_args[@]}" "${HOMEBREW_CELLAR}" || exit 1)"
    if [[ -n "${formula_output}" ]]
```
Same below.

Reply:
@carlocab This doesn't work as expected because it's running in a sub-shell. Already tried this 😁